### ISSUE TITLE
@Schema(hidden = true) doesn't work when @JsonProperty in @JsonCreator is present

### ISSUE THREAD

**Issue by joschi** [swagger-api/swagger-core#3697]
When a field has been hidden with `@Schema(hidden = true)`, the resulting OpenAPI specification will still include the field as write-only if there is a parameter in the `@JsonCreator` method annotated with `@JsonProperty` for the respective property:

```java
public class ModelWithHiddenFieldsInJsonCreator {
    @JsonProperty("id")
    private final Long id;

    @io.swagger.v3.oas.annotations.media.Schema(hidden = true)
    @JsonProperty("hidden")
    private final String hidden;

    @JsonCreator
    public ModelWithHiddenFieldsInJsonCreator(@JsonProperty("id") Long id,
                                              @io.swagger.v3.oas.annotations.media.Schema(hidden = true)
                                              @JsonProperty("hidden") String hidden) {
        this.id = id;
        this.hidden = hidden;
    }

    public Long getId() {
        return id;
    }

    @io.swagger.v3.oas.annotations.media.Schema(hidden = true)
    public String getHidden() {
        return hidden;
    }
}
```

This will still show the property "hidden" in the OpenAPI specification, although I'd expect the property to be hidden.

**Comment by tGuichin** [comment]
Not sure how ```@Schema(hidden = true)``` works, but if I need to hide any field in Swagger, I use ```@Parameter(hidden = true)``` or if I need to hide an endpoint I would use ```@Operation(hidden = true)```

**Comment by joschi** [comment]
@tGuichin `@Parameter(hidden = true)` also doesn't work.

I get the same result with the following class as with the class in my original comment:
```java
public class ModelWithHiddenFieldsInJsonCreator {
    @JsonProperty("id")
    private final Long id;

    @Parameter(hidden = true)
    @JsonProperty("hidden")
    private final String hidden;

    @JsonCreator
    public ModelWithHiddenFieldsInJsonCreator(@JsonProperty("id") Long id,
                                              @JsonProperty("hidden") @Parameter(hidden = true) String hidden) {
        this.id = id;
        this.hidden = hidden;
    }

    public Long getId() {
        return id;
    }

    @Parameter(hidden = true)
    public String getHidden() {
        return hidden;
    }
}
```

**Comment by frantuma** [comment]
Thanks for reporting this! fixed in #3706, closing ticket, please reopen if still experiencing issues
